examples/gno.land/r/demo/art/haiku/haiku.gno

```diff
@@ -0,0 +1,381 @@
+package haiku
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"regexp"
+	"std"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/grc/grc721"
+	"gno.land/p/demo/ufmt"
+	"gno.land/r/demo/users"
+)
+
+var (
+	// URL for the realm
+	pageURL = "/r/demo/art/haiku"
+	// regex for sanitizer
+	nonAlphanumericRegex = regexp.MustCompile(`[^a-zA-Z ]+`)
+	// definitions for the NFT
+	nftName   = "TextNFT"
+	nftSymbol = "NFTEXT"
+	// storage for the fnts
+	nfts *grc721.BasicNFT
+	// storage for the syllable dictionary
+	wordsWithSyllables avl.Tree
+	// storage for the haikus
+	nftDB avl.Tree // tokenID (string) -> haikuAmulet
+)
+
+// errors
+var (
+	ErrNotEnglish          error  = errors.New("non-english word detected")
+	ErrIncorrectLineNumber error  = errors.New("incorrect number of lines")
+	ErrIncorrectSyllables  error  = errors.New("incorrect number of syllables")
+	StatusNotFound         string = "404"
+)
+
+// init initializes the NFT, the database, and the dictionary
+// of syllables
+func init() {
+	nfts = grc721.NewBasicNFT(nftName, nftSymbol)
+	nftDB = avl.Tree{}
+	// load dicitionary of words
+	wordsWithSyllables = avl.Tree{}
+	for _, line := range strings.Split(rawwordlist, "\n") {
+		line = strings.TrimSpace(line)
+		word := strings.Replace(line, ";", "", -1)
+		wordsWithSyllables.Set(word, strings.Count(line, ";")+1)
+	}
```

Comment:
I think we should encourage offchain codegen for such needs.

It also reminds me what @albttx is doing with merkle tree based airdrops.

Reply:
Another option is to work on a pager system to initialize a contract with a lot of data:

```go
// func init()

var paused = true

func InitBatch(batch string, done bool) {
  assertIsAdmin()
  handleBatch(batch)
  if done {
    paused = false
  }
}

func PublicMethod() {
  assertIsNotPaused()
  // ...
}
```

Reply:
Hi @moul, I've been going through some iterations on this and haven't really enabled a big improvement yet. 

> offchain codegen for such needs.

I'm not sure I am grokking this solution - what do you mean by offchain here? Is there an example to follow?

> Another option is to work on a pager

Yeah - this option makes sense - if I understand correctly I can just load up the wordlist from the calling `gno maketx call` with an authorized call to add words...that solution works, but I feel like it would miss out on one of the gno.land benefits - that the source code is available to publicly view (unless there is a way to view a `avl.Tree` of a realm?). In this example, it is nice that the wordlist is publicly available so it is completely transparent about the inclusion rules to submit poems.